flight/PiOS/posix/pios_sys.c

```diff
@@ -239,7 +319,12 @@ static int handle_device(const char *optarg) {
 		bmm150_cfg = PIOS_malloc(sizeof(*bmm150_cfg));
 		bzero(bmm150_cfg, sizeof(*bmm150_cfg));
 
-		bmm150_cfg->orientation = PIOS_BMM_TOP_90DEG;
+		if (mag_orientation == 255) {
+			bmm150_cfg->orientation = PIOS_BMM_TOP_90DEG;
+		} else {
+			/* XXX figure out orientation properly */
```

Comment:
Yah.  This is kinda a weird case (specified an explicit mag orientation, but using the flyingpi's mag).  So I'm not entirely sure what's right.